### PR TITLE
Refactor New Tab / Highlight selectors

### PR DESCRIPTION
**Pull request by k88hudson**
This patch simplifies the selectors used for Highlights by:

- Renaming "Spotlight" to "Highlights"
- putting some of the old stuff related to pre-weighted highlights into isolated, separate selectors (in `oldSpotlightSelectors.js`)
- simplifying deduping code (with a refactor of `selectAndDedupe`)
- adding detailed comments for most selectors

### CODE REVIEW

File: content-src/selectors/oldSpotlightSelectors.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const firstRunData = require("lib/first-run-data");
```
Comment (k88hudson) [inline content-src/selectors/oldSpotlightSelectors.js]: this was moved from `selectors.js`

**Comment by k88hudson** [comment]
@sarracini would you mind taking a look?

**Comment by sarracini** [comment]
@k88hudson yup I'll review!

File: content-src/selectors/selectors.js [hunk 2]
```diff
@@ -1,144 +1,68 @@
...
 module.exports.selectNewTabSites = createSelector(
   [
-    selectWeightedHighlights,
     state => state.Experiments.values.weightedHighlights,
-    selectTopSites,
-    state => state.History,
-    selectSpotlight,
+    state => (state.Experiments.values.weightedHighlights ? state.WeightedHighlights : selectSpotlight(state)),
+    state => state.TopSites,
+    state => (state.Experiments.values.weightedHighlights ? Object.assign({}, state.History, {rows: []}) : state.History),
```
Comment (sarracini) [inline content-src/selectors/selectors.js]: just to be sure i understand: this is saying if the weighted highlights are on, we don't want to even populate the history selector (aka recent items)

File: content-src/selectors/selectors.js [hunk 3]
```diff
@@ -1,144 +1,68 @@
...
+      {sites: History.rows}
+    ]);
+
+    if (!isNewHighlights) {
+      // Find the index of the recommendation. If we have an index, find that recommendation and put it
+      // in the third highlights spot
+      const recommendation = highlightsRows.findIndex(element => element.recommended);
+      if (recommendation >= 0) {
+        highlightsRows.splice(2, 0, highlightsRows.splice(recommendation, 1)[0]);
+      }
```
Comment (sarracini) [inline content-src/selectors/selectors.js]: you know what, just remove this while you're here - we're not showing recommendations and the logic will most likely change for showing recommendations anyways so feel free to take this out

File: content-src/selectors/selectors.js (hunk 2 above)
Comment (k88hudson) [inline content-src/selectors/selectors.js]: that's right (because the new design doesn't include history at all)

**Pull request merged**